### PR TITLE
feat(ui): attribute data type with params

### PR DESCRIPTION
**Pull request by martinmaillard**
Allows defining attribute data type and parameters from the UI.

### CODE REVIEW

File: ui/src/datatypes.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  { name: 'normalizedString', uri: expand('xsd:normalizedString'), params: defaultParams },
+  { name: 'token', uri: expand('xsd:token'), params: defaultParams },
+  { name: 'language', uri: expand('xsd:language'), params: defaultParams },
+  { name: 'Name', uri: expand('xsd:Name'), params: defaultParams },
+  { name: 'NMTOKEN', uri: expand('xsd:NMTOKEN'), params: defaultParams },
+  { name: 'xml', uri: expand('rdf:xml'), params: defaultParams },
+  { name: 'html', uri: expand('rdf:html'), params: defaultParams },
+  { name: 'json', uri: expand('csvw:json'), params: defaultParams },
+  { name: 'time', uri: expand('xsd:time'), params: [...defaultParams, 'format'] }
+]
```
Comment (tpluscode) [inline ui/src/datatypes.ts]: 😲

File: ui/src/types.ts [hunk 2]
```diff
@@ -66,6 +66,21 @@ export interface TableFormData {
   sourceId: ResourceId,
 }
 
+export type DataTypeParam = 'format' | 'default' | 'language'
```
Comment (tpluscode) [inline ui/src/types.ts]: Language is not a datatype param.

And on second inspection turns out that neither is `default` 😞

Comment (martinmaillard) [reply ui/src/types.ts]: It's not in CSVW, but it is in the UI (just because it makes sense)

**Comment by martinmaillard** [comment]
> You should not have moved language inside the datatype.

Care to explain why?

From my understanding, language is definitely part of the datatype (even if it's not defined like that in the spec).
- When it is defined, it means the value is implicitly a string (even though the "string" type is a different one)
- It cannot be defined with any other type

I guess "default" is arguable.

File: ui/src/datatypes.ts (hunk 1 above)
Comment (martinmaillard) [reply ui/src/datatypes.ts]: We'll probably want to cherry-pick the types we want to show to the user. As far as I know, some of these will never be used, and others are even redundant (int vs integer)...

Comment (tpluscode) [reply ui/src/datatypes.ts]: Does seems a little too csvw-centric how what we've been trying to be agnostic. Fine for now but it's just the observation

**Comment by tpluscode** [comment]
All right. o de we agree to keep `language` as tied to the datatype but move `default` up, to the attribute itself?

**Comment by martinmaillard** [comment]
I extracted `default` in the API call, but I'm keeping it in the data type params in the UI for now. Not 100% sure about this, but I think it's more practical.

The whole datatype thing will probably have to be improved after we use it for a while. If we only support `format` long term, I will probably show all the fields instead of hiding them. If we decide to support more params, we'll have to think about it.

File: ui/src/datatypes.ts (hunk 1 above)
Comment (martinmaillard) [reply ui/src/datatypes.ts]: Yeah I agree, but it seemed like the pragmatic thing to do. I think?

**Pull request merged**